src/meta/src/manager/session_params.rs

```diff
@@ -0,0 +1,185 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::ops::{Deref, DerefMut};
+use std::sync::Arc;
+
+use risingwave_common::session_config::{SessionConfig, SessionConfigError};
+use thiserror_ext::AsReport;
+use tokio::sync::RwLock;
+use tracing::info;
+
+use crate::model::{ValTransaction, VarTransaction};
+use crate::storage::{MetaStore, MetaStoreRef, Snapshot, Transaction};
+use crate::{MetaError, MetaResult};
+
+pub type SessionParamsManagerRef = Arc<SessionParamsManager>;
+
+pub struct SessionParamsManager {
```

Comment:
Please add some rustdocs. This struct actually manages the "global session params", which might be confusing for other folks.